File: Kevin van den Bekerom/DSLQL/src/main/java/org/uva/sea/ql/ast/expr/LEq.java
package org.uva.sea.ql.ast.expr;

import org.uva.sea.ql.ast.form.Context;
import org.uva.sea.ql.ast.visit.Visitor;
import org.uva.sea.ql.type.NumericalType;
import org.uva.sea.ql.type.Type;

public class LEq extends BinaryExpr {
	
	public LEq(Expr lhs, Expr rhs) {
		super.lhs = lhs;
		super.rhs = rhs;
	}
	
	@Override
	public Boolean eval() {
		return (Integer) lhs.eval() <= (Integer) rhs.eval();
	}
	
	public void accept(Visitor visitor, Object context) {
		visitor.visit(this, context);
	}

	@Override
	public String toString() {
		return super.toString() + " (<=)";
	}
	
	@Override
	public Type getType(Context context) {
		return new NumericalType();
	}
}

File: Edward Poot/src/ql/ast/expr/binaryoperatorexpr/leq.go
package binaryoperatorexpr

import (
	"ql/ast/expr"
)

type LEq struct {
	Lhs, Rhs expr.Expr
}

func (l LEq) Eval(s interface{}) interface{} {
	return l.Lhs.Eval(s).(int) <= l.Rhs.Eval(s).(int)
}